Spigot-API-Patches/0285-add-isDeeplySleeping-to-HumanEntity.patch

```diff
@@ -0,0 +1,26 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Jake Potrebic <jake.m.potrebic@gmail.com>
+Date: Thu, 8 Apr 2021 17:36:15 -0700
+Subject: [PATCH] add isDeeplySleeping to HumanEntity
+
+
+diff --git a/src/main/java/org/bukkit/entity/HumanEntity.java b/src/main/java/org/bukkit/entity/HumanEntity.java
+index f0e0710fef5a3e0b722ece7ccf89c3d0f88f8f0f..2ce774c81a93260a1464183d435b4c418ed61648 100644
+--- a/src/main/java/org/bukkit/entity/HumanEntity.java
++++ b/src/main/java/org/bukkit/entity/HumanEntity.java
+@@ -319,6 +319,15 @@ public interface HumanEntity extends LivingEntity, AnimalTamer, InventoryHolder
+      */
+     public void setCooldown(@NotNull Material material, int ticks);
+ 
++    // Paper start
++    /**
++     * If the player has slept enough to count towards passing the night.
```

Comment:
Maybe should mention that this will return true when `HumanEntity#getSleepTicks` >= `100`

Reply:
I think the point of adding this is so people don’t have to rely on a magic number (100) to know if a player is deeply sleeping or not.

Reply:
I don't see how that's relying on a magic number, it's just documenting the behavior of the method, if the number changes we can/should update the doc. Developers should be calling the method, not reading the doc and then re implementing it based on it's description.